### DISCUSSION TITLE
Allow Remix application to omit package.json (e.g. when working with nx executors)

### DISCUSSION THREAD

**Discussion by mcmxcdev** [remix-run/remix#3348] [Proposals]
At the moment, it is not possible to move all npm scripts into Nx-specific `project.json` and delete the `package.json`.

Remix is throwing an error like:
`ENOENT: no such file or directory, open '/home/<user>/<project>/package.json'`

Could there be an exemption to not throw the error when working with workspace tools like NX?

**Comment by kiliman** [comment]
Even in an `Nx` project, wouldn't the individual project require a *project.json* file? I thought `Nx` simply coordinates builds and consolidates `node_modules`.

With that being said, Remix needs the _package.json_ dependencies in order to know what packages need to be bundled by the client build.

https://github.com/remix-run/remix/blob/4b11c6d12309ba5a1f3be4f716739f3240f21c35/packages/remix-dev/compiler.ts#L318-L327

Anyway, you should be able to move the scripts. Just note that the `remix` command is executed from `node_modules/.bin/remix`, so make sure `Nx` is executing the correct command.

**Reply by mcmxcdev** [reply]
What we want to achieve is a `package.json` at root level only, and each subpackage has a nx-specific `project.json`, no `package.json`.

All dependencies including remix ones moved to root level, so the nested `package.json` files are essentially useless.
